python/dask_cudf/dask_cudf/groupby.py

```diff
@@ -314,6 +316,15 @@ def groupby_agg(
         for col in aggs:
             _aggs[col] = _aggs[col][0]
     _meta = ddf._meta.groupby(gb_cols, as_index=as_index).agg(_aggs)
+    if aggs_renames:
+        col_array = []
+        agg_array = []
+        for col, agg in _meta.columns:
+            col_array.append(col)
+            agg_array.append(
+                aggs_renames.get(_make_name(col, agg, sep=sep), agg)
+            )
+        _meta.columns = pd.MultiIndex.from_arrays([col_array, agg_array])
```

Comment:
I don't like that we have to do the aggregation renames for both `_meta` _and_ the groupby result, but this is required so that we have the correct `final_columns` for the last step of `_finalize_gb_agg()`. It would be nice if we also supported nested dict aggregations in cuDF's groupby so that `_meta` would have the correct index without any additional steps in dask-cuDF.

Reply:
I think @shwina said this could be done but would require some effort.  Since pandas does not support nested dicts it seemed like cuDF did not have to go down this path.  We could be wrong and if you feel strongly you should speak up

Reply:
>Since pandas does not support nested dicts it seemed like cuDF did not have to go down this path

If pandas doesn't support something ugly, I'd lean away from doing it in cudf for the sake of dask-cudf logic :)

Reply:
Yeah I generally agree - there could be larger motivations to want nested renaming support for groupby in cuDF, but I don't think this case alone is a good enough reason to work on it